external/fv3fit/fv3fit/keras/_models/shared/halos.py

```diff
@@ -178,24 +178,25 @@ def append_halos_tensor(n_halo: int, tensor: tf.Tensor) -> tf.Tensor:
 def _append_halos_tensor(n_halo: int, tensor: tf.Tensor) -> tf.Tensor:
     comms = _create_comms(total_ranks=6)
     communicators = [_get_cubed_sphere_communicator(comm) for comm in comms]
-    shape = list(tensor.shape[1:])
-    shape[0] += 2 * n_halo
+    # need to split tensors into one for each tile
+    shape = [tensor.shape[0]] + list(tensor.shape[2:])
     shape[1] += 2 * n_halo
+    shape[2] += 2 * n_halo
     quantities = [
         pace.util.Quantity(
             data=np.zeros(shape, dtype=_dtype_to_numpy[tensor.dtype]),
-            dims=["x", "y", "z"],
+            dims=["sample", "x", "y", "z"],
             units="unknown",
-            origin=(n_halo, n_halo, 0),
-            extent=tuple(tensor.shape[1:]),
+            origin=(0, n_halo, n_halo, 0),
+            extent=tuple([tensor.shape[0]] + list(tensor.shape[2:])),
         )
         for _ in range(6)
     ]
     for i, quantity in enumerate(quantities):
-        quantity.view[:] = tensor[i, :, :, :].numpy()
+        quantity.view[:] = tensor[:, i, :].numpy()
```

Comment:
It's tricky for me to verify if this is the expected change. How good is your test coverage here? If you introduce a fault on purpose e.g. flip the index order incorrectly does it break some test?